feed/tests/test_comment_views.py

```diff
@@ -0,0 +1,83 @@
+import pytest
+from feed.models import Post, Comment
+from django.contrib.auth.models import User
+
+
+USERNAME1 = 'username1'
+USERNAME2 = 'username2'
+USER_PASS = 'secret_pass'
+POST_TITLE = 'post title'
+POST_CONTENT = 'some post content'
+COMMENT_CONTENT = 'some comment content'
+LOGIN_URL = '/login/'
+REDIRECT_URL_STATUS = 302
+PAGE_NOT_FOUND_STATUS = 404
+
+
+def new_comment_url(post_parent_id):
+    return f"/post/{post_parent_id}/comment/new/"
+
+
+def post_detail_view_url(post_id):
+    return f"/post/{post_id}/"
+
+
+def comment_delete_url(comment_id):
+    return f"/comment/{comment_id}/delete/"
+
+
+@pytest.fixture
+def users(db):
+    user1 = User.objects.create_user(USERNAME1, password=USER_PASS)
+    user2 = User.objects.create_user(USERNAME2, password=USER_PASS)
+    return [user1, user2]
+
+
+@pytest.fixture
+def post(db, users):
+    post = Post.posts.create(title=POST_TITLE, content=POST_CONTENT, author=users[0])
+    post.save()
+    return post
+
+
+@pytest.fixture
+def comment(db, post, users):
+    comment = Comment.comments.create(post_parent=post, content=COMMENT_CONTENT, author=users[1])
+    comment.save()
+    return comment
+
+
+@pytest.fixture
+def logged_in_client(client, users):
+    client.force_login(users[0])
+    return client
+
+
+@pytest.mark.django_db
+class TestCommentDeleteView:
+    def test_delete_url_for_logged_out_user(self, comment, client):
+        # Testing to verify that when logged out user tryies to delete a comment
+        # the user gets a 302 response code and gets redirected to login page
+        # Also checks for the existence of the comment
+        initial_comment_no = Comment.comments.count()
+        response = client.get(comment_delete_url(comment.id))
+        assert response.status_code == REDIRECT_URL_STATUS
+        assert response.url == f'{LOGIN_URL}?next={comment_delete_url(comment.id)}'
+        assert Comment.comments.count() == initial_comment_no
+
+    def test_delete_url_for_non_author_user(self, logged_in_client, comment):
+        # Testing that if a logged in, non author, user tries to delete
+        # a comment, a 403 response will return
+        # also, making sure the comment wasn't delted
+        initial_comment_no = Comment.comments.count()
+        response = logged_in_client.get(comment_delete_url(comment.id))
+        assert response.status_code == 403
+        assert Comment.comments.count() == initial_comment_no
+
+    def test_success_delete_using_delete_view(self, client, request, comment, users):
+        # Testing successfully deleting a comment using the delete view via the delete comment URL
+        # Making sure the author of the post is logged in
+        initial_comment_no = Comment.comments.count()
+        client.force_login(users[1])
+        client.post(comment_delete_url(comment.id))
```

Comment:
You already have the comment.id from the fixture
You can have fixture for initial_comment_no

You could save the coomant.id, and try to get it through client, isnt it?
You are doing part of it in L73

Reply:
Added a fixture for initial_comment_no.
I cant get the comment ID from client since the comment never appears on the client side in this process, the user only get a confirmation form and I submit it via `client.post(...)`